salt/vault/roles.sls

```diff
@@ -1,22 +1,12 @@
 {% set roles = salt.pillar.get('vault:roles') %}
 {% for role_id, role in roles.items() %}
-{% for environment in role.environments %}
-{% set env_data = salt.pillar.get('environments:{}'.format(environment), {}) %}
-  {% for purpose in env_data.purposes %}
-create_{{ role.name }}_role_in_{{ role.backend }}_for_{{ purpose }}_in_{{ environment }}:
+create_{{ role_id }}:
```

Comment:
role_id is the arbitrary keypath in the pillar file that is purely for the purpose of differentiating between other entries in the pillar file. role.name is the actual role name that will be used as part of the URL for retrieving the credentials (e.g. `mysql-mitx-qa/creds/admin`)